### ISSUE TITLE
Replace OpenCover in favor of coverlet.console for .NET Core unit tests

### ISSUE THREAD

**Issue by AdmiringWorm** [cake-contrib/Cake.Recipe#570]
This is something that would be nice to have, as coverlet works cross-platform and is more nicely integrated .NET Core than OpenCover is.

Unfortunately, there are some issues that not entirely prevent the implementation, but rather includes an undesirable workaround to make it work.

The reason behind this is because of what seems to be a design decision in the addin `Cake.Coverlet`, the path it expects for the DLL is hard-coded to use a Debug Configuration rather than being configurable (I have not reached out to the maintainer yet to see if this can be changed).
The workaround would then be that we will need to manually use the `StartProcess` cake alias to invoke the coverlet.console utility ourselves, which is not the desired approach at this time.

This is still something we would like to tackle some time in the future, but may not happen until an eventual 3.x release.

**Comment by AdmiringWorm** [comment]
Cross-reference https://github.com/Romanx/Cake.Coverlet/issues/34 .
This issue is currently blocked by the above linked issue.

**Comment by AdmiringWorm** [comment]
Once the next version of Cake.Coverlet hits, this issue can be resolved as well.

Would really like to get it into 2.0, but we will see.

**Comment by AdmiringWorm** [comment]
Situations have changed regarding this issue, instead of completely replacing OpenCover, instead Coverlet will be preferred for .NET Core builds before OpenCover is run.

Additionally, .NET Framework builds can opt-in to run Coverlet tool instead of OpenCover as well.
Due to the changes that were necessary, it was easier to implement all logic without relying on the Cake.Coverlet addin, and the addin itself will be removed as well.

What to use can be changed by the user (coverlet msbuild, coverlet collector and opencover for .NET core , and coverlet console and opencover for .NET Framework).
